check/controls.go

```diff
@@ -79,12 +74,82 @@ func NewControls(in []byte, definitions []string) (*Controls, error) {
 	return c, nil
 }
 
+func (controls *Controls) convertAuditToRegisteredType(auditType AuditType, audit interface{}) (auditer Auditer, err error) {
+
+	var auditBytes  []byte
+	if auditType == "" || auditType == TypeAudit {
+		if s, ok := audit.(string); ok || audit == nil {
+
+			return Audit(s), nil
+		}
+		return nil, fmt.Errorf("failed to convert audit, mismatching type")
+	} else {
+		if callback, ok := controls.auditTypeRegistry[auditType]; ok {
+			o := callback()
+			if auditBytes, err = yaml.Marshal(audit); err == nil {
+				if err := yaml.Unmarshal(auditBytes, o); err != nil {
+					return nil, fmt.Errorf("unable to Unmarshal Audit %v", err)
+				}
+				return o.(Auditer), nil // we don't check error because it already verified in convertAuditToRegisteredType
```

Comment:
@igorbeider I think you can remove this comment because now you are checking the error